src/main/java/org/opensearch/ad/rest/handler/ModelValidationActionHandler.java

```diff
@@ -0,0 +1,718 @@
+                            this
+                        );
+                    // this case means all intervals up to max interval recommendation length have been tried
+                    // which further means the next step is to go through A/B validation checks
+                } else {
+                    intervalListener.onResponse(null);
+                }
+
+            } catch (Exception e) {
+                onFailure(e);
+            }
+        }
+
+        @Override
+        public void onFailure(Exception e) {
+            logger.error("Failed to paginate top anomaly results", e);
+            listener.onFailure(e);
+        }
+    }
+
+    private void processIntervalRecommendation(IntervalTimeConfiguration interval, long latestTime) {
+        if (interval == null) {
+            checkRawDataSparsity(latestTime);
+        } else {
+            if (interval.equals(anomalyDetector.getDetectionInterval())) {
+                logger.info("Using the current interval there is enough dense data ");
+                // The rate of buckets with at least 1 doc with given interval is above the success rate
+                listener.onResponse(null);
+                return;
+            }
+            // return response with interval recommendation
+            listener
+                .onFailure(
+                    new ADValidationException(
+                        CommonErrorMessages.DETECTOR_INTERVAL_REC + interval.getInterval(),
+                        DetectorValidationIssueType.DETECTION_INTERVAL,
+                        ValidationAspect.MODEL,
+                        interval
+                    )
+                );
+        }
+    }
+
+    private AggregationBuilder getBucketAggregation(long latestTime) {
+        return AggregationBuilders
+            .dateHistogram("agg")
+            .field(anomalyDetector.getTimeField())
+            .minDocCount(0)
+            .hardBounds(getTimeRangeBounds(latestTime, (IntervalTimeConfiguration) anomalyDetector.getDetectionInterval()))
+            .fixedInterval(DateHistogramInterval.minutes((int) anomalyDetector.getDetectorIntervalInMinutes()));
+    }
+
+    private SearchSourceBuilder getSearchSourceBuilder(QueryBuilder query, AggregationBuilder aggregation) {
+        return new SearchSourceBuilder().query(query).aggregation(aggregation).size(0).timeout(requestTimeout);
+    }
+
+    private void checkRawDataSparsity(long latestTime) {
+        AggregationBuilder aggregation = getBucketAggregation(latestTime);
+        SearchSourceBuilder searchSourceBuilder = new SearchSourceBuilder().aggregation(aggregation).size(0).timeout(requestTimeout);
+        try {
+            AggregatorFactories.Builder internalAgg = parseAggregators(
+                anomalyDetector.getFeatureAttributes().get(0).getAggregation().toString(),
+                xContentRegistry,
+                anomalyDetector.getFeatureAttributes().get(0).getId()
+            );
```

Comment:
Shouldn't this include all of the features and not just the `0`th one?

Reply:
This shouldn't actually be even added to the raw data aggregation and I removed it.